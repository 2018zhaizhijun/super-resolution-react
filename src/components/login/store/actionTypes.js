export const CHANGE_IS_VISIBLE = 'login/CHANGE_IS_VISIBLE';
//export const CHANGE_REGISTER_STATE = 'login/CHANGE_REGISTER_STATE';
export const CHANGE_LOGIN_STATE = 'login/CHANGE_USER_LOGIN_STATE';
export const CHANGE_AVATAR = 'login/CHANGE_AVATAR';
export const CHANGE_PROFILE = 'login/CHANGE_PROFILE_INFO';
export const CHANGE_TOKEN = 'login/CHANGE_PROFILE_TOKEN';
