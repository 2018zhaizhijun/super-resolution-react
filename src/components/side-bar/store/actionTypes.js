export const CHANGE_IS_VISIBLE = 'sidebar/CHANGE_IS_VISIBLE';
