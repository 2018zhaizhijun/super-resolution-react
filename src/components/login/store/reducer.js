import { Map } from 'immutable';
import * as actionTypes from './actionTypes';

const defaultState = Map({
    isVisible: false,
    //registering: false,
    isLogin: false,
    profile: {
        avatarUrl: null,
        usrname: null,
        telephone: null,
        sex: 0,
    },
    token: null,
});

function reducer(state=defaultState, action) {
    switch (action.type) {
        case actionTypes.CHANGE_IS_VISIBLE:
            return state.set('isVisible', action.payload);
        // case actionTypes.CHANGE_REGISTER_STATE:
        //     return state.set('registering', action.registering);
        case actionTypes.CHANGE_LOGIN_STATE:
            return state.set('isLogin', action.payload);
        // case actionTypes.CHANGE_AVATAR:
        //     return state.set('avatarUrl', action.avatarUrl);
        case actionTypes.CHANGE_PROFILE:
            return state.set('profile', {...state.get('profile'), ...action.payload});
        case actionTypes.CHANGE_TOKEN:
            return state.set('token', action.payload);
        default:
            return state;
    }
}

export default reducer;