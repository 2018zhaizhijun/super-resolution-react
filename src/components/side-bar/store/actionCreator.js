import * as actionTypes from './actionTypes';
import { createAction } from '@reduxjs/toolkit';

export const changeIsVisible = createAction(actionTypes.CHANGE_IS_VISIBLE);
// export const changeIsVisible = (visibleState) => ({
//     type: actionTypes.CHANGE_IS_VISIBLE,
//     isVisible: visibleState
// });
