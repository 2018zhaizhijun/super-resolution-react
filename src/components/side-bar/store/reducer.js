import { Map } from 'immutable';
import * as actionTypes from './actionTypes';

const defaultState = Map({
    isVisible: false,
});

function reducer(state=defaultState, action) {
    switch (action.type) {
        case actionTypes.CHANGE_IS_VISIBLE:
            return state.set('isVisible', action.payload);
        default:
            return state;
    }
}

export default reducer;