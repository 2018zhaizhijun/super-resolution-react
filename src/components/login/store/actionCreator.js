import { phoneLogin, getAvatar, getProfile, logout, sendAvatar } from '@service/login';
import * as actionTypes from './actionTypes';
import { message } from 'antd';
import { createAction } from '@reduxjs/toolkit';

import { sendUsrinfo } from '@service/login';

// 用createAction创建action可以不用再对其额外写单元测试
export const changeIsVisible = createAction(actionTypes.CHANGE_IS_VISIBLE);
// export const changeIsVisible = (visibleState) => ({
//     type: actionTypes.CHANGE_IS_VISIBLE,
//     isVisible: visibleState
// });

// export const changeRegisterState = (registerState) => ({
//     type: actionTypes.CHANGE_IS_VISIBLE,
//     registering: registerState
// });

export const changeLoginState = createAction(actionTypes.CHANGE_LOGIN_STATE);
// export const changeLoginState = (loginState) => ({
//     type: actionTypes.CHANGE_LOGIN_STATE,
//     isLogin: loginState
// });

// export const changeAvatar = (avatarUrl) => ({
//     type: actionTypes.CHANGE_AVATAR,
//     avatarUrl
// });

export const changeProfile = createAction(actionTypes.CHANGE_PROFILE);
// export const changeProfile = (profile) => ({
//     type: actionTypes.CHANGE_PROFILE,
//     profile
// });

export const changeToken = createAction(actionTypes.CHANGE_TOKEN);
// export const changeToken = (token) => ({
//     type: actionTypes.CHANGE_TOKEN,
//     token
// });

export const performLogin = (telephone, password) => {
    return (dispatch) => {
        return phoneLogin(telephone, password).then((response) => {
            console.log(response);
            if (response.statusCode != 200) {
                message.error('账号或密码错误');
            }
            else {
                const token = response.data.token;
                dispatch(changeLoginState(true));
                dispatch(changeToken(token));
                dispatch(changeIsVisible(false));

                return getProfile();  
            }
        }).then((res) => {
            console.log(res);
            if (res.statusCode != 200) {
                message.error('用户信息获取失败');
            }
            else {
                let data = res.data;
                dispatch(changeProfile({
                    avatarUrl: data.avatar,
                    usrname: data.trueName,
                    telephone,
                    sex: data.sex,
                }));
            }
        }).catch((e) => {
            message.error('登录失败');
            console.error('phoneLoginError', e);
        })
    };
};

export const performLogout = () => {
    return (dispatch) => {
        return logout().then( (res) => {
            if (res.statusCode == 200) {
                dispatch(changeLoginState(false));
            }
            else {
                throw new Error('statusCode' + res.statusCode);
            }
        }).catch((e) => {
            message.error('退出登录失败');
            console.error('logoutError', e);
        })
    };
};

// export const updateAvatar = (avatarFile) => {
//     return (dispatch) => {
//         sendAvatar(avatarFile).then( (res) => {
//             if (res.statusCode == 200) {
//                 dispatch(changeAvatar(res.data.avatarUrl));
//             }
//             else {
//                 message.error('头像上传失败');
//             }
//         }).catch((e) => console.error('sendAvatarError', e.message))
//     };
// };
