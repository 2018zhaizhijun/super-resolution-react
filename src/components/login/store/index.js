import reducer from './reducer';
import { changeIsVisible, changeProfile, performLogin, performLogout } from './actionCreator';

export {
    reducer,
    changeIsVisible,
    changeProfile,
    performLogin,
    performLogout,
    //updateAvatar,
}