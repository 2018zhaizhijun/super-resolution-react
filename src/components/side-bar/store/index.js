import reducer from './reducer';
import { changeIsVisible as changeSidebarVisible } from './actionCreator';

export {
    reducer,
    changeSidebarVisible,
}