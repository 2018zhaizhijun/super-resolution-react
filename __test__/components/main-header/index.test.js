import React from 'react';
import MainHeader from '@components/main-header';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import store from '@store';
import { Map } from 'immutable';
import { mount } from 'enzyme';
import toJson from "enzyme-to-json";

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

const loginStore = mockStore(Map({
    login: {
        isLogin: true,
        profile: {
            avatarUrl: null,
            usrname: 'zz',
            telephone: '12345678910',
            sex: 1,
        }
    },
    sidebar: {
        isVisible: false
    }
}));

describe('main header render', () => {

  it('main header without login should match snapshot', () => {
    const headerWrapper = renderer.create(
        <Provider store={store}>
            <BrowserRouter>
                <MainHeader />
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(headerWrapper).toMatchSnapshot();
  });

  it('main header with login should match snapshot', () => {
    const headerWrapper = mount(
        <Provider store={loginStore}>
            <BrowserRouter>
                <MainHeader />
            </BrowserRouter>
        </Provider>
    );
    headerWrapper.find('.drop-menu').simulate('mouseover');
    expect(toJson(headerWrapper)).toMatchSnapshot();
  });

  it('different navigation buttons should lead to correct url path', () => {
    const wrapper = mount(
        <Provider store={loginStore}>
            <BrowserRouter>
                <MainHeader />
            </BrowserRouter>
        </Provider>
    );

    expect(wrapper.find('#toTransPage').first().prop('to'))
        .toEqual('/main/transform');

    expect(wrapper.find('#toIntroPage').first().prop('to'))
        .toEqual('/main/discribe');

    //const test = wrapper.find('.drop-menu');
    const dropMenu = wrapper.find('Dropdown').prop('overlay')();
    const dropWrapper = mount(
        <BrowserRouter>
            { dropMenu }
        </BrowserRouter>
    );
    expect(dropWrapper.find('#toUsrinfoPage').first().prop('to'))
        .toEqual('/main/usrInfo');
    expect(dropWrapper.find('#toHistPage').first().prop('to'))
        .toEqual('/main/history');
  })

})