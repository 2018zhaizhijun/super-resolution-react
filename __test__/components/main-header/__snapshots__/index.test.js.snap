// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main header render main header with login should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Memo(MainHeader)>
        <ErrorBoundary>
          <styled.div>
            <div
              className="sc-bdvvtL TCZnK"
            >
              <div
                className="logo-menu"
              >
                <img
                  alt="logo"
                  className="logo"
                  src={Object {}}
                />
              </div>
              <div
                className="nav-left"
              >
                <NavLink
                  activeClassName="active-link"
                  className="header-link"
                  id="toTransPage"
                  key="超分转换"
                  to="/main/transform"
                >
                  <Link
                    aria-current={null}
                    className="header-link"
                    id="toTransPage"
                    to={
                      Object {
                        "hash": "",
                        "pathname": "/main/transform",
                        "search": "",
                        "state": null,
                      }
                    }
                  >
                    <LinkAnchor
                      aria-current={null}
                      className="header-link"
                      href="/main/transform"
                      id="toTransPage"
                      navigate={[Function]}
                    >
                      <a
                        aria-current={null}
                        className="header-link"
                        href="/main/transform"
                        id="toTransPage"
                        onClick={[Function]}
                      >
                        <i>
                          超分转换
                        </i>
                      </a>
                    </LinkAnchor>
                  </Link>
                </NavLink>
                <NavLink
                  activeClassName="active-link"
                  className="header-link"
                  id="toIntroPage"
                  key="功能简介"
                  to="/main/discribe"
                >
                  <Link
                    aria-current={null}
                    className="header-link"
                    id="toIntroPage"
                    to={
                      Object {
                        "hash": "",
                        "pathname": "/main/discribe",
                        "search": "",
                        "state": null,
                      }
                    }
                  >
                    <LinkAnchor
                      aria-current={null}
                      className="header-link"
                      href="/main/discribe"
                      id="toIntroPage"
                      navigate={[Function]}
                    >
                      <a
                        aria-current={null}
                        className="header-link"
                        href="/main/discribe"
                        id="toIntroPage"
                        onClick={[Function]}
                      >
                        <i>
                          功能简介
                        </i>
                      </a>
                    </LinkAnchor>
                  </Link>
                </NavLink>
              </div>
              <div
                className="usr-menu"
              >
                <Dropdown
                  mouseEnterDelay={0.15}
                  mouseLeaveDelay={0.1}
                  overlay={[Function]}
                >
                  <ForwardRef(Dropdown)
                    mouseEnterDelay={0.15}
                    mouseLeaveDelay={0.1}
                    overlay={[Function]}
                    overlayClassName=""
                    placement="bottomLeft"
                    prefixCls="ant-dropdown"
                    transitionName="ant-slide-up"
                  >
                    <Trigger
                      action={
                        Array [
                          "hover",
                        ]
                      }
                      afterPopupVisibleChange={[Function]}
                      autoDestroy={false}
                      blurDelay={0.15}
                      builtinPlacements={
                        Object {
                          "bottomCenter": Object {
                            "offset": Array [
                              0,
                              4,
                            ],
                            "overflow": Object {
                              "adjustX": 1,
                              "adjustY": 1,
                            },
                            "points": Array [
                              "tc",
                              "bc",
                            ],
                            "targetOffset": Array [
                              0,
                              0,
                            ],
                          },
                          "bottomLeft": Object {
                            "offset": Array [
                              0,
                              4,
                            ],
                            "overflow": Object {
                              "adjustX": 1,
                              "adjustY": 1,
                            },
                            "points": Array [
                              "tl",
                              "bl",
                            ],
                            "targetOffset": Array [
                              0,
                              0,
                            ],
                          },
                          "bottomRight": Object {
                            "offset": Array [
                              0,
                              4,
                            ],
                            "overflow": Object {
                              "adjustX": 1,
                              "adjustY": 1,
                            },
                            "points": Array [
                              "tr",
                              "br",
                            ],
                            "targetOffset": Array [
                              0,
                              0,
                            ],
                          },
                          "topCenter": Object {
                            "offset": Array [
                              0,
                              -4,
                            ],
                            "overflow": Object {
                              "adjustX": 1,
                              "adjustY": 1,
                            },
                            "points": Array [
                              "bc",
                              "tc",
                            ],
                            "targetOffset": Array [
                              0,
                              0,
                            ],
                          },
                          "topLeft": Object {
                            "offset": Array [
                              0,
                              -4,
                            ],
                            "overflow": Object {
                              "adjustX": 1,
                              "adjustY": 1,
                            },
                            "points": Array [
                              "bl",
                              "tl",
                            ],
                            "targetOffset": Array [
                              0,
                              0,
                            ],
                          },
                          "topRight": Object {
                            "offset": Array [
                              0,
                              -4,
                            ],
                            "overflow": Object {
                              "adjustX": 1,
                              "adjustY": 1,
                            },
                            "points": Array [
                              "br",
                              "tr",
                            ],
                            "targetOffset": Array [
                              0,
                              0,
                            ],
                          },
                        }
                      }
                      defaultPopupVisible={false}
                      destroyPopupOnHide={false}
                      focusDelay={0}
                      getDocument={[Function]}
                      getPopupClassNameFromAlign={[Function]}
                      hideAction={Array []}
                      mask={false}
                      maskClosable={true}
                      mouseEnterDelay={0.15}
                      mouseLeaveDelay={0.1}
                      onPopupAlign={[Function]}
                      onPopupVisibleChange={[Function]}
                      overlay={[Function]}
                      popup={[Function]}
                      popupAlign={Object {}}
                      popupClassName=""
                      popupPlacement="bottomLeft"
                      popupStyle={Object {}}
                      popupTransitionName="ant-slide-up"
                      prefixCls="ant-dropdown"
                      showAction={Array []}
                      stretch="minWidth"
                    >
                      <div
                        className="ant-dropdown-trigger drop-menu"
                        data-testid="dropMenu"
                        key="trigger"
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                      >
                        <img
                          alt=""
                          src={Object {}}
                        />
                        <ForwardRef(DownOutlined)
                          className="icon"
                        >
                          <AntdIcon
                            className="icon"
                            icon={
                              Object {
                                "icon": Object {
                                  "attrs": Object {
                                    "focusable": "false",
                                    "viewBox": "64 64 896 896",
                                  },
                                  "children": Array [
                                    Object {
                                      "attrs": Object {
                                        "d": "M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z",
                                      },
                                      "tag": "path",
                                    },
                                  ],
                                  "tag": "svg",
                                },
                                "name": "down",
                                "theme": "outlined",
                              }
                            }
                          >
                            <span
                              aria-label="down"
                              className="anticon anticon-down icon"
                              role="img"
                            >
                              <IconReact
                                icon={
                                  Object {
                                    "icon": Object {
                                      "attrs": Object {
                                        "focusable": "false",
                                        "viewBox": "64 64 896 896",
                                      },
                                      "children": Array [
                                        Object {
                                          "attrs": Object {
                                            "d": "M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z",
                                          },
                                          "tag": "path",
                                        },
                                      ],
                                      "tag": "svg",
                                    },
                                    "name": "down",
                                    "theme": "outlined",
                                  }
                                }
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  key="svg-down"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                    key="svg-down-svg-0"
                                  />
                                </svg>
                              </IconReact>
                            </span>
                          </AntdIcon>
                        </ForwardRef(DownOutlined)>
                      </div>
                    </Trigger>
                  </ForwardRef(Dropdown)>
                </Dropdown>
              </div>
            </div>
          </styled.div>
          <Memo(Login)>
            <Modal
              confirmLoading={false}
              footer={null}
              okType="primary"
              onCancel={[Function]}
              title="Super Resolution"
              visible={false}
              width={520}
            >
              <Dialog
                closeIcon={
                  <span
                    className="ant-modal-close-x"
                  >
                    <ForwardRef(CloseOutlined)
                      className="ant-modal-close-icon"
                    />
                  </span>
                }
                confirmLoading={false}
                focusTriggerAfterClose={true}
                footer={null}
                maskTransitionName="ant-fade"
                okType="primary"
                onCancel={[Function]}
                onClose={[Function]}
                prefixCls="ant-modal"
                title="Super Resolution"
                transitionName="ant-zoom"
                visible={false}
                width={520}
                wrapClassName=""
              >
                <PortalWrapper
                  visible={false}
                />
              </Dialog>
            </Modal>
          </Memo(Login)>
        </ErrorBoundary>
      </Memo(MainHeader)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`main header render main header without login should match snapshot 1`] = `
<div
  className="sc-bdvvtL TCZnK"
>
  <div
    className="logo-menu"
  >
    <img
      alt="logo"
      className="logo"
      src={Object {}}
    />
  </div>
  <div
    className="nav-left"
  >
    <a
      aria-current={null}
      className="header-link"
      href="/main/transform"
      id="toTransPage"
      onClick={[Function]}
    >
      <i>
        超分转换
      </i>
    </a>
    <a
      aria-current={null}
      className="header-link"
      href="/main/discribe"
      id="toIntroPage"
      onClick={[Function]}
    >
      <i>
        功能简介
      </i>
    </a>
  </div>
  <div
    className="usr-menu"
  >
    <div
      className="login-link"
      onClick={[Function]}
    >
      登录
    </div>
  </div>
</div>
`;
